lib/place_mark.rb

```diff
@@ -23,6 +21,14 @@ def execute(player:, x:, y:)
   private
 
   def check_dup?(x, y)
-    @board[x][y] == :O || @board[x][y] == :X
+    coordinate_has_O?(x,y) || coordinate_has_X?(x,y)
+  end
```

Comment:
A shorter way to write this could be  [:O, :X].include? (@board[x][y]) , instead of having check_dup, coordinate_has_O and coordinate_has_X.